.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1262,43 +1262,63 @@ private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject refe
             var ret = new DateTimeResolutionResult();
 
             var er = this.config.DateExtractor.Extract(text, referenceDate);
-            if (er.Count < 2)
-            {
-                er = this.config.DateExtractor.Extract(this.config.TokenBeforeDate + text, referenceDate);
-                if (er.Count < 2)
-                {
-                    return ret;
-                }
-
-                er[0].Start -= this.config.TokenBeforeDate.Length;
-                er[1].Start -= this.config.TokenBeforeDate.Length;
-            }
 
-            var match = this.config.WeekWithWeekDayRangeRegex.Match(text);
-            string weekPrefix = null;
-            if (match.Success)
-            {
-                weekPrefix = match.Groups["week"].ToString();
+            var pr1 = new DateTimeParseResult();
+            var pr2 = new DateTimeParseResult();
+
+            // Handle with "now"
+            var nowMatches = this.config.NowRegex.Matches(text);
```

Comment:
We shouldn't have to even do this, if the previous Extract already returned two entities.

Reply:
And everything related to "now", including this regex match should go into the method.